app/models/twitch_user.rb

```diff
@@ -0,0 +1,59 @@
+require 'twitch'
+
+class TwitchUser < ApplicationRecord
+  belongs_to :user
+
+  def self.from_auth(auth)
+    twitch_user = TwitchUser.find_or_initialize_by(twitch_id: auth.uid)
+    twitch_user.user = User.new if twitch_user.user.blank?
+    twitch_user.user.name = auth.info.nickname
+    twitch_user.assign_attributes(
+      access_token: auth.credentials.token,
+      name:         auth.info.nickname,
+      display_name: auth.info.name,
+      email:        auth.info.email,
+      avatar:       auth.info.image || TwitchUser.default_avatar
+    )
+    self.errors.add(twitch_user.user.errors) unless twitch_user.user.save
+    twitch_user
+  end
+
+  def avatar
+    URI.parse(self[:avatar]).tap do |uri|
+      uri.scheme = 'https'
+    end.to_s
+  end
+
+  def sync!
+    body = JSON.parse(Twitch::User.get(twitch_id))
+
+    update(
+      twitch_id:    body['_id'],
+      name:         body['name'].downcase,
+      display_name: body['display_name'],
+      avatar:       URI.parse(body['logo'] || self.class.default_avatar).tap { |uri| uri.scheme = 'https' }.to_s
+    )
+  rescue RestClient::ResourceNotFound
+    nil
+  end
+
+  def sync_follows!
+    ActiveRecord::Base.transaction do
+      current_followed_users = User.where(twitch_id: Twitch::Follows.followed_ids(twitch_id))
+
+      # If TwitchUserFollow is changed to have child records or destroy callbacks, change this to destroy_all
+      TwitchUserFollow.where(from_user: user, to_user: (twitch_followed_users - current_followed_users)).delete_all
```

Comment:
This job now throws an error on `twitch_followed_users` since this is no longer in the `User` model, and it is a method defined over there.